### PR TITLE
avoid reset float array size

### PR DESCRIPTION
**Pull request by zhakesi**

### CODE REVIEW

File: native/cocos/2d/renderer/RenderDrawInfo.cpp [hunk 1]
```diff
@@ -32,9 +32,9 @@
 namespace cc {
 
 static gfx::DescriptorSetInfo gDsInfo;
-cc::Float32Array& mat4ToFloat32Array(const cc::Mat4& mat, cc::Float32Array& out, index_t ofs = 0) {
-    memcpy(reinterpret_cast<float*>(const_cast<uint8_t*>(out.buffer()->getData())) + ofs, mat.m, 16 * sizeof(float));
-    return out;
+static float matrixData[pipeline::UBOLocal::COUNT];
```
Comment (dumganhar) [inline native/cocos/2d/renderer/RenderDrawInfo.cpp]: It's better to initialize variable explictly.
```c++
static float matrixData[pipeline::UBOLocal::COUNT] = {0.F};
```

File: native/cocos/2d/renderer/RenderDrawInfo.cpp [hunk 2]
```diff
@@ -156,11 +156,9 @@ void RenderDrawInfo::updateLocalDescriptorSet(Node* transform, gfx::DescriptorSe
     }
     _localDSBF->ds->bindBuffer(pipeline::UBOLocal::BINDING, _localDSBF->uboBuf);
     _localDSBF->ds->update();
-    static Float32Array matrixData;
-    matrixData.reset(pipeline::UBOLocal::COUNT);
     const auto& worldMatrix = transform->getWorldMatrix();
-    mat4ToFloat32Array(worldMatrix, matrixData, pipeline::UBOLocal::MAT_WORLD_OFFSET);
-    _localDSBF->uboBuf->update(matrixData.buffer()->getData());
+    mat4ToFloatArray(worldMatrix, matrixData, pipeline::UBOLocal::MAT_WORLD_OFFSET);
+    _localDSBF->uboBuf->update(matrixData);
```
Comment (dumganhar) [inline native/cocos/2d/renderer/RenderDrawInfo.cpp:161]: Could avoid memcpy?
```c++
_localDSBF->uboBuf->update(worldMatrix.m, sizeof(Mat4));
```

Comment (dumganhar) [reply native/cocos/2d/renderer/RenderDrawInfo.cpp:161]: Then, no need to add a mat4ToFloat32Array  function.

Comment (dumganhar) [reply native/cocos/2d/renderer/RenderDrawInfo.cpp:161]: OK, I see the offset. Ignore previous comment.

**Review by dumganhar — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**